dashboard/views.py

```diff
@@ -321,60 +326,69 @@ def resource_access_within_week(request, course_id=0):
-
     # return if there is no data during this interval
     if (df.empty):
         return HttpResponse("{}")
 
-    # group by resource_id, and resource_name
-    # reformat for output
-    df['resource_id_name'] = df['resource_id'].astype(str).str.cat(df['resource_name'], sep=';')
-
-    df=df.drop(['resource_id', 'resource_name'], axis=1)
-    df.set_index(['resource_id_name'])
+    df.set_index(['resource_id_type'])
     # drop resource records when the resource has been accessed multiple times by one user
     df.drop_duplicates(inplace=True)
 
     # map point grade to letter grade
     df['grade'] = df['current_grade'].map(gpa_map)
 
     # calculate the percentage
-    df['percent'] = df.groupby(['resource_id_name', 'grade'])['resource_id_name'].transform('count') / total_number_student
+    df['percent'] = df.groupby(['resource_id_type', 'grade'])['resource_id_type'].transform('count') / total_number_student
 
     df=df.drop(['current_grade', 'user_id'], axis=1)
     # now only keep the resource access stats by grade level
     df.drop_duplicates(inplace=True)
 
-    resource_id_name=df["resource_id_name"].unique()
+
+    resource_id_type=df["resource_id_type"].unique()
 
     #df.reset_index(inplace=True)
 
     # zero filled dataframe with resource name as row name, and grade as column name
-    output_df=pd.DataFrame(0.0, index=resource_id_name, columns=[GRADE_A, GRADE_B, GRADE_C, GRADE_LOW, NO_GRADE_STRING, RESOURCE_TYPE_STRING])
-    output_df=output_df.rename_axis('resource_id_name')
+    output_df=pd.DataFrame(0.0, index=resource_id_type, columns=['r_id', 'r_name', GRADE_A, GRADE_B, GRADE_C, GRADE_LOW, NO_GRADE_STRING, RESOURCE_TYPE_STRING])
+    output_df=output_df.rename_axis('resource_id_type')
     output_df=output_df.astype({RESOURCE_TYPE_STRING: str})
+    output_df=output_df.astype({'r_name': str})
+    output_df=output_df.astype({'r_id': str})
+
 
     for index, row in df.iterrows():
         # set value
-        output_df.at[row['resource_id_name'], row['grade']] = row['percent']
-        output_df.at[row['resource_id_name'], RESOURCE_TYPE_STRING] = row[RESOURCE_TYPE_STRING]
+        output_df.at[row['resource_id_type'], row['grade']] = row['percent']
+        output_df.at[row['resource_id_type'], RESOURCE_TYPE_STRING] = row[RESOURCE_TYPE_STRING]
+        output_df.at[row['resource_id_type'], 'r_name'] = row['name']
+        output_df.at[row['resource_id_type'], 'r_id'] = row['resource_id']
     output_df.reset_index(inplace=True)
 
     # now insert person's own viewing records: what resources the user has viewed, and the last access timestamp
-    selfSqlString = f"""select CONCAT(r.resource_id, ';', r.name) as resource_id_name, count(*) as self_access_count, max(a.access_time) as self_access_last_time 
+    selfSqlString = f"""
+                    select
+                    r.resource_id as resource_id,
+                    r.resource_type as resource_type,
+                    r.name as name,
```

Comment:
Could also do the same id CONCAT here too instead of Pandas. Then wouldn't need to drop the resource_type later.

Reply:
done!